pkg/dynamicinstrumentation/testutil/e2e_test.go

```diff
@@ -0,0 +1,188 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf && arm64
+
+package testutil
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"reflect"
+	"strings"
+	"testing"
+	"text/template"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation"
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation/diconfig"
+	"github.com/DataDog/datadog-agent/pkg/dynamicinstrumentation/ditypes"
+	"github.com/cilium/ebpf"
+	"github.com/cilium/ebpf/features"
+	"github.com/cilium/ebpf/rlimit"
+	"github.com/kr/pretty"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestGoDI(t *testing.T) {
+	if err := rlimit.RemoveMemlock(); err != nil {
+		t.Error(err)
+	}
+
+	if features.HaveMapType(ebpf.RingBuf) != nil {
+		t.Skip("ringbuffers not supported on this kernel")
+	}
+
+	sampleServicePath := BuildSampleService(t)
+	cmd := exec.Command(sampleServicePath)
+	cmd.Env = []string{
+		"DD_DYNAMIC_INSTRUMENTATION_ENABLED=true",
+		"DD_SERVICE=go-di-sample-service",
+		"DD_DYNAMIC_INSTRUMENTATION_OFFLINE=true",
+	}
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
```

Comment:
You may want to send the program output to `t.Log` if you care about having it with your test results.
```suggestion
	// send stdout to t.Log
	stdoutPipe, err := cmd.StdoutPipe()
	require.NoError(t, err)
	go func() {
		scanner := bufio.Scanner(stdoutPipe)
		for scanner.Scan() {
			t.Log(scanner.Text())
		}
		if err := scanner.Err(); err != nil {
			t.Log(err)
		}
	}()
	// send stderr to stdout pipe
	cmd.Stderr = cmd.Stdout
```